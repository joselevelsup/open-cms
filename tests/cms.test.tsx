import * as React from "react";
import { mount } from "enzyme";
import CmsPage from "../src/components/cmspage";

describe("CmsPage", () => {
	it("CmsPage Renders with only required props", () => {
		const wrapper = mount(
			<CmsPage otherRoutes={[{ name: "home page", apiRoute: "/home"}]} apiRoute={"/home"} />
		);

		expect(wrapper).toMatchSnapshot();
	});
});
