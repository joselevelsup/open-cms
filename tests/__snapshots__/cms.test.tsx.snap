// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsPage CmsPage Renders with only required props 1`] = `
<styled.div
  className="cms-page"
>
  <styled.div
    className="cms-header"
    logo={null}
  >
    <div
      className="route-link"
      key="0"
    >
      <a
        href="/admin/home-page"
      >
        home page
      </a>
    </div>
  </styled.div>
  <br />
  <div
    className="cms-page-options"
  >
    <Styled(styled.button)
      className="cms-option"
    >
      Cancel
    </Styled(styled.button)>
    <Styled(styled.button)
      className="cms-option"
      onClick={[Function]}
    >
      Load CMS Data
    </Styled(styled.button)>
    <Styled(styled.button)
      className="cms-option"
      onClick={[Function]}
      updateAlert={false}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <p>
           Save Changes
        </p>
      </div>
    </Styled(styled.button)>
  </div>
  <styled.div
    className="cms-body"
  >
    <div
      className="components"
    >
      <div
        className="component-item"
        key="0"
      >
        <p>
          short text
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="1"
      >
        <p>
          long text
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="2"
      >
        <p>
          nested
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="3"
      >
        <p>
          media
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="4"
      >
        <p>
          link
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
    </div>
    <div
      className="main"
    />
  </styled.div>
</styled.div>
`;

exports[`CmsPage should render with only required props 1`] = `
<styled.div
  className="cms-page"
>
  <styled.div
    className="cms-header"
    logo={null}
  >
    <div
      className="route-link"
      key="0"
    >
      <a
        href="/admin/home-page"
      >
        home page
      </a>
    </div>
  </styled.div>
  <br />
  <div
    className="cms-page-options"
  >
    <Styled(styled.button)
      className="cms-option"
    >
      Cancel
    </Styled(styled.button)>
    <Styled(styled.button)
      className="cms-option"
      onClick={[Function]}
    >
      Load CMS Data
    </Styled(styled.button)>
    <Styled(styled.button)
      className="cms-option"
      onClick={[Function]}
      updateAlert={false}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <p>
           Save Changes
        </p>
      </div>
    </Styled(styled.button)>
  </div>
  <styled.div
    className="cms-body"
  >
    <div
      className="components"
    >
      <div
        className="component-item"
        key="0"
      >
        <p>
          short text
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="1"
      >
        <p>
          long text
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="2"
      >
        <p>
          nested
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="3"
      >
        <p>
          media
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
      <div
        className="component-item"
        key="4"
      >
        <p>
          link
        </p>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <IoMdAdd
            color="white"
            style={
              Object {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </Styled(styled.button)>
      </div>
    </div>
    <div
      className="main"
    />
  </styled.div>
</styled.div>
`;
